src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -1640,30 +1645,26 @@ type RideHailingEstimate {
     productName: String
 }
 
-type Leg {
-    """
-    The date and time when this leg begins. Format: Unix timestamp in milliseconds.
-    """
-    startTime: Long
+"""
+An ISO-8601-formatted datetime with offset, i.e. `2023-06-13T14:30+03:00` for 2:30pm on June 13th 2023 at Helsinki's offset from UTC at that time.
+"""
+scalar OffsetDateTime
 
+type LegTimes {
```

Comment:
`ComplexTime` ?
`Call` ?

Reply:
Is there some potential feature of a leg time that differs from a "stop time"? Also, does it make sense to always return this type which includes realtime related information even when it's not really feasible that there is any. For example if you walk from origin to a bike rental station? Maybe we should return an union of "realtime times" and "non-realtime times". Are there any other potential use cases for realtime times other than transit? Maybe we should call the realtime version "TransitTime" or "ComplexTime" and the other "SimpleTime"? For some reason, nothing feels perfect. Also, using a union/interface makes the client maybe slightly more complicated.

Reply:
I think the best solution would be to make `Leg` an interface and the implementations define what kind of time it provides.

But we all know that we can only do that with a new schema.

I will be doing some thinking if a union is better or not.